R/xp_utils.R

```diff
@@ -11,32 +11,59 @@ xp_text_in_table <- function(table) {
   return(paste0("text() = ", table, collapse = " or "))
 }
 
-# convert an XML match into a Lint
+#' Convert an XML match into a Lint
+#'
+#' Convenience function for converting nodes matched by XPath-based
+#'   linter logic into a [Lint()] object to return.
+#'
+#' @inheritParams Lint
+#' @param xml An `xml_node` object (to generate one `Lint`) or an
+#'   `xml_nodeset` object (to generate several `Lint`s), e.g. as returned by
+#'   [xml2::xml_find_all()] or [xml2::xml_find_first()].
+#' @param source_expression A source expression object, e.g. as
+#'   returned by [get_source_expressions()].
+#' @param lint_message The message to be included as the `message`
+#'   to the `Lint` object. If `lint_message` is a `function`,
+#'   this function is first applied to `xml` (so it should be a
+#'   function taking an `xml_node` as input and must produce a
+#'   length-1 character as output).
+#' @param offset Integer, default 0. The amount by which to offset the
+#'   `col1` and `col2` values taken from `xml` when producing
+#'   the `ranges` value in the `Lint` object.
+#' @export
 xml_nodes_to_lint <- function(xml, source_expression, lint_message,
                               type = c("style", "warning", "error"),
-                              offset = 0L,
-                              global = FALSE) {
+                              offset = 0L) {
+  if (length(xml) == 0L) {
+    return(list())
+  }
+  if (inherits(xml, "xml_nodeset")) {
+    return(lapply(xml, xml_nodes_to_lint, source_expression, lint_message, type, offset))
```

Comment:
Maybe class it as `Lints` for nicer printing?

Reply:
doesn't `lint.R` handle that?

IIUC this matches the behavior at HEAD of `lapply(expr, xml_nodes_to_lint)`, right?

Reply:
Yes, but if we export it, I see how it might be used for interactive development of the XPaths, where the nicer print() might be cool.